test/modules/DirectMethodSender/ReporterClients/TestResultCoordinatorReporterClient.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.ModuleUtil.ReporterClients
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public sealed class TestResultReporterClient : ReporterClientBase
+    {
+        ILogger logger;
+        Uri baseUri;
+        TestResultReportingClient testResultReportingClient = null;
+        TestResultReportingClient TestResultReportingClient
+        {
+            get
+            {
+                if (this.testResultReportingClient == null)
+                {
+                    Preconditions.CheckNotNull(this.baseUri, nameof(this.baseUri));
+                    this.testResultReportingClient = new TestResultReportingClient { BaseUrl = this.baseUri.AbsoluteUri };
+                    return this.testResultReportingClient;
+                }
+                else
+                {
+                    return this.testResultReportingClient;
+                }
+            }
+        }
+
+        internal TestResultReporterClient(Uri baseUri, ILogger logger)
```

Comment:
I remember default constructor access modifier is internal, you don't need to specify here.